### ISSUE TITLE
Latest myget builds pull in System.Runtime for .NET 4.5

### ISSUE THREAD

**Issue by thebigb** [dotnet/efcore#2755]
When you create an empty _.NET 4.5_ console application project and attempt to install the package `EntityFramework7.SqlServer`, NuGet will attempt to install the `System.Runtime` package, as shown below. This breaks the build process, and if you attempt to fix it creates a horrible dependency mess. This the case for at least _beta6-13753_ to the latest one (_beta7-13816_).

I believe it's `System.Collections.Immutable` that's the culprit here (but I'm not 100% sure, NuGet tooling does not provide a lot of insight on dependency resolution). I've noticed that since _1.1.37_ it adds some dependencies for `.NETPlatform5.0`.

From my perspective it seems that either:
- `System.Collections.Immutable` is not mapping the _.NET 5.0_ dependencies properly
- NuGet is broken, and grabbing the wrong dependency set
- Some other funky stuff is going on

The problem probably isn't in the EF7 package, but could anyone shed some light on this?

Some version info, if it's any help:

```
Microsoft Visual Studio Community 2015
Version 14.0.23107.0 D14REL

Microsoft .NET Framework
Version 4.6.00079

NuGet Package Manager   3.0.0
```

```
PM> Install-Package EntityFramework7.SqlServer -Version 7.0.0-beta7-13816 -DependencyVersion Lowest -WhatIf
Attempting to gather dependencies information for package 'EntityFramework7.SqlServer.7.0.0-beta7-13816' with respect to project 'EF7', targeting '.NETFramework,Version=v4.5'
Attempting to resolve dependencies for package 'EntityFramework7.SqlServer.7.0.0-beta7-13816' with DependencyBehavior 'Lowest'
Resolving actions to install package 'EntityFramework7.SqlServer.7.0.0-beta7-13816'
Resolved actions to install package 'EntityFramework7.SqlServer.7.0.0-beta7-13816'
Install Ix-Async 1.2.4
Install Microsoft.Framework.Caching.Abstractions 1.0.0-beta7-11640
Install Microsoft.Framework.Configuration.Abstractions 1.0.0-beta7-11610
Install Microsoft.Framework.Configuration 1.0.0-beta7-11610
Install Microsoft.Framework.Configuration.Binder 1.0.0-beta7-11610
Install Microsoft.Framework.DependencyInjection.Abstractions 1.0.0-beta7-11609
Install Microsoft.Framework.DependencyInjection 1.0.0-beta7-11609
Install Microsoft.Framework.Logging.Abstractions 1.0.0-beta7-11614
Install Microsoft.Framework.Logging 1.0.0-beta7-11614
Install Microsoft.Framework.OptionsModel 1.0.0-beta7-11578
Install Microsoft.Framework.Caching.Memory 1.0.0-beta7-11640
Install Remotion.Linq 2.0.0-alpha-004
Install System.Linq 4.0.0
Install System.Resources.ResourceManager 4.0.0
Install System.Runtime 4.0.10-beta-23019
Install System.Collections 4.0.10-beta-22231
Install System.Diagnostics.Debug 4.0.10-beta-22231
Install System.Globalization 4.0.10-beta-22231
Install System.Runtime.Extensions 4.0.10-beta-22231
Install System.Threading.Tasks 4.0.0-beta-23019
Install System.Threading 4.0.10-beta-22605
Install System.Collections.Immutable 1.1.37
Install EntityFramework7.Core 7.0.0-beta7-13816
Install EntityFramework7.Relational 7.0.0-beta7-13816
Install EntityFramework7.SqlServer 7.0.0-beta7-13816
```

Thanks!

**Comment by ErikEJ** [comment]
Looks very similar to https://github.com/aspnet/EntityFramework/issues/2747 - thanks for the detailed investigation

**Comment by thebigb** [comment]
Okay, so the issue appears not to be `System.Runtime` being pulled in, but rather a broken dependency chain within the `System.Runtime` dependency.

I also found out that the stable `System.Collections.Immutable` package and its dependencies do not have this issue. So, if its compatible I would advise to select the stable version of `System.Collections.Immutable` as a dependency.

**Comment by rowanmiller** [comment]
@thebigb are you still using the `EntityFramework7.*` packages? We swapped back to ` EntityFramework.*` (dropping the 7) a while back, so just want to make sure you have the latest packages.

**Comment by thebigb** [comment]
@rowanmiller I did come across two discussion threads, but apparently skimmed over them too quickly.

Anyway, I just tried the latest `EntityFramework.*` packages on MyGet and they appear to suffer from the same issue.

**Comment by ErikEJ** [comment]
Indeed...

**Comment by bricelam** [comment]
Some of the .NET Core packages RTMed on July 29th. We're currently in the process of updating our dependencies. I'll see if this still repros once that work is in.

**Comment by bricelam** [comment]
It looks like one of our dependencies now requires a minimum of .NET 4.6, but otherwise this issue appears to be resolved. Can you confirm?

**Comment by thebigb** [comment]
The packages now install without failure, but I immediately get the following exception while running:

`Could not load file or assembly 'System.Collections, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (Exception from HRESULT: 0x80131058)`

Removing the `<runtime></runtime>` section from the App.config worked around this (but you should probably check that out, because it's automatically added, and sometimes re-emerges with other package installs).

But I'm also struggling to get inheritance working, and I'm not sure if it's related.

I get an exception with the message `Sequence contains no matching element` on the statement `products.Any()` in the setup below. The same statement works properly with a setup that has no inheritance.

```
class Program
{
	static void Main(string[] args)
	{
		using (var store = new StoreContext())
		{
			var products = store.Set<Service>();

			if (!products.Any())
			{
				store.Add(new Service { Id = Guid.NewGuid().ToString(), PricePerYear = 999.95M });
				store.Add(new Service { Id = Guid.NewGuid().ToString(), PricePerYear = 50M });
				store.Add(new ShelfProduct { Id = Guid.NewGuid().ToString(), PricePerUnit = 5.50M });
				store.Add(new WeighedProduct { Id = Guid.NewGuid().ToString(), PricePerKilogram = 5.05M, WeightInGrams = 1000 });
				store.SaveChanges();
			}
		}
	}
}

public class StoreContext : DbContext
{
	protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
	{
		optionsBuilder.UseSqlite("Data Source=sandbox.db");
	}

	protected override void OnModelCreating(ModelBuilder modelBuilder)
	{
		modelBuilder.Entity<Product>();
		modelBuilder.Entity<Service>().BaseType<Product>();
		modelBuilder.Entity<ShelfProduct>().BaseType<Product>();
		modelBuilder.Entity<WeighedProduct>().BaseType<Product>();
	}
}

public abstract class Product
{
	public string Id { get; set; }
}

public class Service : Product
{
	public decimal PricePerYear { get; set; }
}

public class ShelfProduct : Product
{
	public decimal PricePerUnit { get; set; }
}

public class WeighedProduct : Product
{
	public decimal PricePerKilogram { get; set; }
	public int? WeightInGrams { get; set; }
}
```

Stacktrace:

```
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source, Func`2 predicate)
   at Microsoft.Data.Entity.Query.ExpressionVisitors.MaterializerFactory.CreateMaterializer(IEntityType entityType, SelectExpression selectExpression, Func`3 projectionAdder, IRelationalMetadataExtensionProvider relationalExtensions, IQuerySource querySource)
   at Microsoft.Data.Entity.Query.ExpressionVisitors.RelationalEntityQueryableExpressionVisitor.VisitEntityQueryable(Type elementType)
   at Microsoft.Data.Entity.Query.ExpressionVisitors.EntityQueryableExpressionVisitor.VisitConstant(ConstantExpression constantExpression)
   at System.Linq.Expressions.ConstantExpression.Accept(ExpressionVisitor visitor)
   at Microsoft.Data.Entity.Query.ExpressionVisitors.ExpressionVisitorBase.Visit(Expression expression)
   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.ReplaceClauseReferences(Expression expression, IQuerySource querySource)
   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.CompileMainFromClauseExpression(MainFromClause mainFromClause, QueryModel queryModel)
   at Microsoft.Data.Entity.Query.RelationalQueryModelVisitor.CompileMainFromClauseExpression(MainFromClause mainFromClause, QueryModel queryModel)
   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.VisitMainFromClause(MainFromClause fromClause, QueryModel queryModel)
   at Remotion.Linq.Clauses.MainFromClause.Accept(IQueryModelVisitor visitor, QueryModel queryModel)
   at Remotion.Linq.QueryModelVisitorBase.VisitQueryModel(QueryModel queryModel)
   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.VisitQueryModel(QueryModel queryModel)
   at Microsoft.Data.Entity.Query.RelationalQueryModelVisitor.VisitQueryModel(QueryModel queryModel)
   at Microsoft.Data.Entity.Query.EntityQueryModelVisitor.CreateQueryExecutor[TResult](QueryModel queryModel)
   at Microsoft.Data.Entity.Storage.RelationalDatabase.CompileQuery[TResult](QueryModel queryModel)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Data.Entity.Query.CompiledQueryCache.CompileQuery(IDatabase database, MethodInfo compileMethodInfo, Type resultItemType, QueryModel queryModel)
   at Microsoft.Data.Entity.Query.CompiledQueryCache.<>c__7`1.<Execute>b__7_0(Expression q, IDatabase ds)
   at Microsoft.Data.Entity.Query.CompiledQueryCache.GetOrAdd(Expression query, QueryContext queryContext, IDatabase database, Boolean isAsync, Func`3 compiler)
   at Microsoft.Data.Entity.Query.CompiledQueryCache.Execute[TResult](Expression query, IDatabase database, QueryContext queryContext)
   at Microsoft.Data.Entity.Query.EntityQueryProvider.Execute[TResult](Expression expression)
   at System.Linq.Queryable.Any[TSource](IQueryable`1 source)
   at EF7.Program.Main(String[] args) in Program.cs:line 18
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
   at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

Could it be related? Or should I open a new issue for this?

**Comment by ErikEJ** [comment]
Nope, when using the latest daily, I still get references to System.Runtime in my test project, and test discovery is completely broken. Will check if removing app.config entries help. But should System.Runtime be installed as a nuget package at all - or is that expected?

**Comment by ErikEJ** [comment]
@bricelam On my 4.5 product project when updating EntityFramework.Relational I get:
1: A app.config with a bogus runtime directive (re System.RunTime)
2: A refrence to System.RunTime
- removing these 2 artifacts unblocks me on the product project.

**Comment by ErikEJ** [comment]
@bricelam Using Build 13969 and I no longer see any issues with app.config appearing or any references to System.Runtime. As far as I can see, this can be closed now

**Comment by thebigb** [comment]
@ErikEJ Have you tried that on a clean project? I can't reproduce that.

**Comment by ErikEJ** [comment]
@thebigb No, that was on an existing project, will try on a blank 4.5 Console app and post the results here!

**Comment by bricelam** [comment]
I think all our feeds have finally caught up and everything appears to be working for me now too. Feel free to re-open this if you're still seeing issues.

**Comment by thebigb** [comment]
@bricelam 

`EntityFramework.Sqlite 7.0.0-beta8-15162` (and its dependencies) is now doing great on .NET 4.5 and 4.6.

However, `EntityFramework.Commands 7.0.0-beta8-15162` leans on `Microsoft.CodeAnalysis.CSharp`, which fails to even install on .NET 4.5, and only works on .NET 4.6 after deleting the entire `<runtime>` section in `App.config`.

**Comment by bricelam** [comment]
I'm going to look at downgrading a few of our dependencies (including Roslyn). I'll see if it helps that case too.
